dev/Repeater/SelectionModel.cpp

```diff
@@ -647,7 +647,11 @@ void SelectionModel::SelectWithPathImpl(const winrt::IndexPath& index, bool sele
         // If we unselect something, raise event any way, otherwise changedSelection is false
         bool changedSelection = false;
 
-        if (m_singleSelect)
+        // We only need to clear selection by walking the data structure from the beginning when we are in single selection mode
+        // and want to select something.
+        // If we want to unselect something we unselect it directly in TraverseIndexPath below and raise the SelectionChanged event
+        // if required.
+        if (m_singleSelect && select)
```

Comment:
So the root cause of the issue was that deselecting an item did not work correctly?

It might be good to also add an API test for this, since the issue is only manifesting in NavigationView but is caused by SelectionModel.

Reply:
Yes. Here's a longer explanation: 

The issue was that unselecting an item in the SelectionModel was not reported back to the NavigationView: In case of item unselection, we first cleared the selection (ClearSelection()) which does not raise the SelectionChanged event. Inside of the TraverseIndexPath() code block we prepare for raising the event by setting the flag `changedSelection` to true in case we updated a node selection state. However, as we already cleared the selection above, all nodes are already unselected and that flag is not set. This causes the event (SelectionChanged) not being raised in case of item deselection, thus the NavigationView, which listens to this event, won't notice the selection of its SelectionModel changed and will not update its SelectedItem property (set it to null).

I will add a test for the SelectionModel tomorrow.